x-pack/plugins/observability/public/context/date_picker_context.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { createContext, useState, useMemo, useCallback } from 'react';
+import useMount from 'react-use/lib/useMount';
+import { useLocation, useHistory } from 'react-router-dom';
+import { parse } from 'query-string';
+import { fromQuery, ObservabilityPublicPluginsStart, toQuery } from '..';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+import { getAbsoluteTime } from '../utils/date';
+
+export interface DatePickerContextValue {
+  relativeStart: string;
+  relativeEnd: string;
+  absoluteStart?: number;
+  absoluteEnd?: number;
+  refreshInterval: number;
+  refreshPaused: boolean;
+  updateTimeRange: (params: { start: string; end: string }) => void;
+  updateRefreshInterval: (params: { interval: number; isPaused: boolean }) => void;
+  lastUpdated: number;
+}
+
+/**
+ * This context contains the time range (both relative and absolute) and the
+ * autorefresh status of the overview page date picker.
+ * It also updates the URL when any of the values change
+ */
+export const DatePickerContext = createContext({} as DatePickerContextValue);
```

Comment:
Try to provide defaults to all non-nullables, or make them nullable if a default value can't be provided. There's a runtime error which seems related to this as shown below.

https://user-images.githubusercontent.com/2748376/157679402-7a8a4ef1-88f1-4803-925e-431b78fa6371.mov

Reply:
I think the problem is that the `DatePickerProvider` is nowhere in the UX dashboard tree. I'll add it.

Reply:
@afgomez that error is gone, thanks.